mongodb_migrations/cli.py

```diff
@@ -13,69 +13,65 @@
+                except Exception as e:
+                    print("Failed to migrate version: %s" % migration_name)
+                    print("%s" % e.message)
+                    sys.exit(1)
+                print("Succeed to migrate version: %s" % migration_name)
+                self._create_migration(migration_name)
 
-def get_mongo_database(host, port, database):
-    client = pymongo.MongoClient(host=host, port=port)
-    return client[database]
+    def _get_migration_names(self):
+        return self.db.database_migrations.find().sort('migration_datetime', pymongo.DESCENDING)
 
+    def _create_migration(self, migration):
+        self.db.database_migrations.save({'migration_datetime': migration,
+                                     'created_at': datetime.now()})
 
-def create_migration(db, migration):
-    db.database_migrations.save({'migration_datetime': migration,
-                                 'created_at': datetime.now()})
+    def _get_mongo_database(self, host, port, database):
+        client = pymongo.MongoClient(host=host, port=port)
+        return client[database]
 
 
 def main():
-    args = parser.parse_args()
-
-    files = os.listdir(args.migrations)
-    migrations = []
-    for f in files:
-        result = re.match('^\d+\.py$', f)
-        if result:
-            migrations.append(f)
-
-    db = get_mongo_database(args.host, args.port, args.database)
-
-    database_migrations = get_migration_names(db)
-    database_migration_names = [migration['migration_datetime'] for migration in database_migrations]
-    migration_names = ['%s' % migration.split('.')[0] for migration in migrations]
-    if set(database_migration_names) - set(migration_names):
-        print("migrations doesn't match")
-        sys.exit(1)
-    last_migration = None
-    if database_migration_names:
-        last_migration = database_migration_names[0]
-        print("Found previous migrations, last migration is version: %s" % last_migration)
-    else:
-        print("No previous migrations found")
-
-    sys.path.insert(0, args.migrations)
-    for migration_name in migration_names:
-        if not last_migration or migration_name > last_migration:
-            print("Trying to migrate version: %s" % migration_name)
-            try:
-                module = __import__(migration_name)
-                migration_object = module.Migration(host=args.host,
-                                                    port=args.port,
-                                                    database=args.database)
-                migration_object.upgrade()
-            except Exception as e:
-                print("Failed to migrate version: %s" % migration_name)
-                print("%s" % e.message)
-                sys.exit(1)
-            print("Succeed to migrate version: %s" % migration_name)
-            create_migration(db, migration_name)
+    manager = MigrationManager()
+    manager.config = Configuration()
```

Comment:
I'd say we should put line 73 - 76 in a `init` method as every time we initialise `MigrationManager`, we need to do this, why not automate it?